shard-operator-parent/shard-operator-v2/src/test/java/com/redhat/service/smartevents/shard/operator/v2/controllers/ManagedProcessorControllerTest.java

```diff
@@ -0,0 +1,54 @@
+package com.redhat.service.smartevents.shard.operator.v2.controllers;
+
+import javax.inject.Inject;
+
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.redhat.service.smartevents.shard.operator.v2.TestSupport;
+import com.redhat.service.smartevents.shard.operator.v2.resources.ManagedProcessor;
+import com.redhat.service.smartevents.test.resource.KeycloakResource;
+
+import io.fabric8.kubernetes.client.utils.KubernetesResourceUtil;
+import io.javaoperatorsdk.operator.api.reconciler.UpdateControl;
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.kubernetes.client.WithOpenShiftTestServer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@QuarkusTest
+@WithOpenShiftTestServer
+@QuarkusTestResource(value = KeycloakResource.class, restrictToAnnotatedClass = true)
+class ManagedProcessorControllerTest {
+
+    @Inject
+    ManagedProcessorController managedProcessorController;
+
+    @Inject
+    ObjectMapper objectMapper;
+
+    @Test
+    void testCreateNewManagedProcessor() {
+        // Given
+        ManagedProcessor managedProcessor = buildManagedProcessor();
+
+        // When
+        UpdateControl<ManagedProcessor> updateControl = managedProcessorController.reconcile(managedProcessor, null);
+
+        // Then
+        assertThat(updateControl.isUpdateStatus()).isFalse();
+        assertThat(managedProcessor.getStatus().isReady()).isTrue();
+    }
+
+    private ManagedProcessor buildManagedProcessor() {
+        return ManagedProcessor.fromBuilder()
+                .withNamespace(KubernetesResourceUtil.sanitizeName(TestSupport.CUSTOMER_ID))
```

Comment:
@lucamolteni You can use the `NamespaceProvider.getNameSpaceName(String bridgeId)` here.